osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-monitoring.xsd

```diff
@@ -540,4 +540,103 @@
     </xsd:sequence>
   </xsd:complexType>
 
+  <!-- ActualPowerQuality -->
+
+  <xsd:element name="ActualPowerQualityRequest">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:ActualPowerQualityRequestData">
+          <xsd:sequence>
+            <xsd:element name="DeviceIdentification" type="common:DeviceIdentification"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:complexType name="ActualPowerQualityRequestData">
+    <xsd:complexContent>
+      <xsd:extension base="common:Query">
+        <xsd:sequence>
+          <xsd:element name="ProfileType" type="xsd:string"/>
+        </xsd:sequence>
+      </xsd:extension>
+    </xsd:complexContent>
+  </xsd:complexType>
+
+  <xsd:element name="ActualPowerQualityResponse">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:ActualPowerQualityResponseData"></xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:complexType name="ActualPowerQualityResponseData">
+    <xsd:complexContent>
+      <xsd:extension base="common:Response">
+        <xsd:sequence>
+          <xsd:element name="ActualPowerQualityData" type="tns:ActualPowerQualityData"
+                       minOccurs="1"
+                       maxOccurs="1"/>
+        </xsd:sequence>
+      </xsd:extension>
+    </xsd:complexContent>
+  </xsd:complexType>
+
+  <xsd:complexType name="ActualPowerQualityData">
+    <xsd:sequence>
+      <xsd:sequence>
+        <xsd:element name="PowerQualityObjects" type="tns:PowerQualityObjects" minOccurs="1" maxOccurs="1"/>
+        <xsd:element name="ActualValues" type="tns:ActualValues" minOccurs="1" maxOccurs="1"/>
+      </xsd:sequence>
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="PowerQualityObjects">
+    <xsd:sequence>
+      <xsd:element type="tns:PowerQualityObject" name="PowerQualityObject"
+                   minOccurs="0" maxOccurs="unbounded" />
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="PowerQualityObject">
+    <xsd:sequence>
+      <xsd:element type="xsd:string" name="Name" />
+      <xsd:element type="common:OsgpUnitType" name="unit" />
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="ActualValues">
+    <xsd:sequence>
+      <xsd:element type="tns:ActualValue" name="ActualValue"
+                   minOccurs="0" maxOccurs="unbounded" />
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="ActualValue">
+    <xsd:choice maxOccurs="unbounded">
```

Comment:
Should `maxOccurs` be unbounded here? It looks like there should only be one of these, but I might be misunderstanding.

Reply:
maxOccurs="unbounded" generates convenient java code here: List<Object> getStringValueOrDateValueOrFloatValue(). instead of four seperated get methods for each datatype. Should be solved with bindings: choiceContentProperty ?